### PR TITLE
Deprecated warnings fix in MO

### PR DESCRIPTION
**Pull request by evolosen**
Solution: Fix DeprecationWarnings

Ticket: 55830
Warning example: model_optimizer/mo/front/tf/partial_infer/tf.py:111: DeprecationWarning: invalid escape sequence \d
  if not match('.*:\d+', out_tensor_name):

Code:
* [x]  Comments - N/A
* [x]  Code style (PEP8) - N/A
* [x]  Transformation generates reshape-able IR - N/A
* [x]  Transformation preserves original framework node names - N/A

Validation:
* [x]  Unit tests - N/A
* [x]  Framework operation tests - N/A
* [x]  Transformation tests - N/A
* [x]  Model Optimizer IR Reader check - N/A

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

**Review by achetver — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]

**Review by yekruglov — APPROVED** [review]

File: model-optimizer/mo/front/tf/loader.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def get_output_node_names_list(graph_def, user_defined_output_node_names_list: l
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/tf/loader.py:140]: What about file `model-optimizer/extensions/back/SpecialNodesFinalization.py`, line 82?

File: model-optimizer/mo/front/tf/loader.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def get_output_node_names_list(graph_def, user_defined_output_node_names_list: l
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/tf/loader.py:141]: What about `model-optimizer/mo/utils/guess_framework.py` lines 31-44?

File: model-optimizer/mo/front/tf/loader.py (hunk 1 above)
Comment (evolosen) [reply model-optimizer/mo/front/tf/loader.py:140]: what are there? I dont see warnings there.

File: model-optimizer/mo/front/tf/loader.py (hunk 2 above)
Comment (evolosen) [reply model-optimizer/mo/front/tf/loader.py:141]: fixed

File: model-optimizer/mo/front/tf/loader.py (hunk 1 above)
Comment (lazarevevgeny) [reply model-optimizer/mo/front/tf/loader.py:140]: My bad. There is already "r" in the beginnig.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**